x-pack/plugins/reporting/server/core.ts

```diff
@@ -82,6 +82,8 @@ export class ReportingCore {
 
   public getContract: () => ReportingSetup;
 
+  private kibanaShuttingDown$ = new Rx.ReplaySubject<void>(1);
```

Comment:
I think `Rx.BehaviorSubject` would be better, since we only need access to the last event value - not all values from an arbitrary number of events.

Reply:
I was looking at `BehaviourSubject` initially, but according to the types and docs it requires an initial value:

```
A variant of Subject that requires an initial value and emits its current
```

I though the API would be simpler with ReplaySubject because it can be instantiated as empty non-emitting stream.